framework/validators/UniqueValidator.php

```diff
@@ -170,7 +170,7 @@ private function getTargetClass($model)
      */
     private function modelExists($targetClass, $conditions, $model)
     {
-        /** @var ActiveRecordInterface $targetClass $query */
+        /* @var $targetClass $query ActiveRecordInterface */
```

Comment:
This should be split into two blocks, separate for `$targetClass` and `$query`?

Reply:
I can remove $targetClass from vardoc and add to @param string|ActiveRecordInterface $targetClass.

Reply:
That would be incorrect - `$targetClass` needs to be a string.

Reply:
No, look at 131 and 137 lines.
```php
$db = $targetClass::getDb();

...

$db->useMaster(function () use ($targetClass, $conditions, $model, &$modelExists) {
    $modelExists = $this->modelExists($targetClass, $conditions, $model);
});
```

$targetClass can be string or ActiveRecordInterface.

Reply:
No, it cannot:

```php
get_class($model) !== $targetClass
```

This will always return `true` if you pass object instead of class name as string.

Reply:
I don't see a problem if modelExists() docblock will contain 2 types. But we can separate the annotation into 2 lines or just remove it.

Reply:
> I don't see a problem if modelExists() docblock will contain 2 types.

The problem is pretty obvious for me - method will not work correctly if you pass `ActiveRecordInterface` instance as `$targetClass`, so adding `ActiveRecordInterface` to phpdoc is misleading.